lib/clients/AzureClient.py

```diff
@@ -18,11 +18,15 @@ def __init__(self, operation_name, configuration, directory_persistent, director
         super(AzureClient, self).__init__(operation_name, configuration, directory_persistent, directory_work_list,
                                           poll_delay_time, poll_maximum_time)
         self.subscription_id = configuration['subscription_id']
-        self.__azureCredentials = ServicePrincipalCredentials(
-            client_id=configuration['client_id'],
-            secret=configuration['client_secret'],
-            tenant=configuration['tenant_id']
-        )
+        if not 'credhub_url' in configuration:
+            self.__setCredentials(
+                configuration['client_id'], configuration['client_secret'], configuration['tenant_id'])
+        else:
+            self.logger.info('fetching creds from credhub')
+            credentials = self._get_credentials_from_credhub(configuration)
```

Comment:
Same as above. What happens if there is an error in getting these credentials? Do we need to raise exception explicitly?